src/pages/docs/v3/modifiers/index.mdx

```diff
@@ -0,0 +1,83 @@
+---
+navigationLabel: Modifiers
+order: 4
+---
+
+# Modifiers
+
+A `modifier` is an object (middleware or plugin) that modifies the `x` and `y`
+coordinates in some way, or provides useful data to the consumer. Popper
+provides the most useful modifiers in its core already for you to import.
+Anything you don't use will be tree-shaken away.
+
+## Example
+
+A very basic modifier that shifts the coordinates by 1 pixel:
+
+```js
+const shiftBy1Px = {
+  // The name of the modifier.
+  name: 'shiftBy1Px',
+  // The main logic which returns new coordinates.
+  fn({ coords }) {
+    return {
+      // Required: always return `x` and `y` coordinates.
+      x: coords.x + 1,
+      y: coords.y + 1,
+
+      // Optional:
+      data: {
+        // Add custom data to be read by other modifiers or the consumer.
+      },
+    };
+  },
+};
+```
+
+### Function
+
+Popper's core modifiers are factory functions, which allows you to pass options
+to configure their behavior:
+
+```js
+const modifier = (options = {}) => ({
+  name: 'modifier',
```

Comment:
Why do we need names?

Reply:
Data objects use them. You _could_ do

```js
return {
  x: number,
  y: number,
  [name]: {...data}
}
```

instead

Reply:
Also, it's possible we might want to add more properties to modifiers later. If we make it just a function that would make the API less clean since we'd be reading off the function object, not a plain one